File: app/src/main/java/me/ycdev/android/devtools/sampler/traffic/AppTrafficStat.java
package me.ycdev.android.devtools.sampler.traffic;

public class AppTrafficStat {
    public int uid;
    public long recvBytes;
    public long sendBytes;

    public AppTrafficStat(int uid) {
        this.uid = uid;
    }

    public long getTotal() {
        return recvBytes + sendBytes;
    }

    public static AppTrafficStat computeUsage(AppTrafficStat oldStat, AppTrafficStat newStat) {
        if (oldStat.uid != newStat.uid) {
            return null;
        }
        AppTrafficStat usage = new AppTrafficStat(newStat.uid);
        usage.recvBytes = newStat.recvBytes - oldStat.recvBytes;
        usage.sendBytes = newStat.sendBytes - oldStat.sendBytes;
        return usage;
    }
}

File: app/src/main/java/me/ycdev/android/devtools/sampler/traffic/AppTrafficStat.kt
package me.ycdev.android.devtools.sampler.traffic

class AppTrafficStat(var uid: Int) {
    var recvBytes: Long = 0
    var sendBytes: Long = 0
    val total: Long
        get() = recvBytes + sendBytes

    companion object {
        fun computeUsage(oldStat: AppTrafficStat, newStat: AppTrafficStat): AppTrafficStat {
            if (oldStat.uid != newStat.uid) {
                throw RuntimeException("Not same pkgName")
            }
            val usage = AppTrafficStat(newStat.uid)
            usage.recvBytes = newStat.recvBytes - oldStat.recvBytes
            usage.sendBytes = newStat.sendBytes - oldStat.sendBytes
            return usage
        }
    }
}
